lib/onetime/app/api/base.rb

```diff
@@ -49,10 +49,14 @@ def authorized allow_anonymous=false
         end
       end
 
-      # Find the locale of the request based on req.env['rack.locale']
+      # Find the locale of the request based on req.env['rack.locale'],
       # which is set automatically by Otto v0.4.0 and greater.
-      # If `locale` is specifies it will override if available.
+      #
+      # If `locale` is specified, it will override if available.
       # If the `local` query param is set, it will override.
+      #
+      # @param locale [String] the locale to use, defaults to nil
```

Comment:
Consider adding type hints to the `check_locale!` method parameters and return value to improve code readability and maintainability. [medium]

Reply:
Created a ticket #309. We'll want to review type hints comprehensively for Ruby 3+. 🤩